src/Jmikola/AutoLogin/Authentication/Token/AutoLoginToken.php

```diff
@@ -54,24 +54,15 @@ public function getCredentials() : string
         return '';
     }
 
-    /**
-     * @see Symfony\Component\Security\Core\Authentication\Token\AbstractToken::serialize()
-     */
-    public function serialize()
+    public function __serialize(): array
     {
-        return serialize(array(
-            $this->key,
-            $this->providerKey,
-            parent::serialize(),
-        ));
+        return [$this->key, $this->providerKey, parent::__serialize()];
     }
 
-    /**
-     * @see Symfony\Component\Security\Core\Authentication\Token\AbstractToken::unserialize()
-     */
-    public function unserialize($str)
+    public function __unserialize(array $data): void
     {
-        [$this->key, $this->providerKey, $parentStr] = unserialize($str);
-        parent::unserialize($parentStr);
+        [$this->key, $this->providerKey, $parentData] = $data;
+        $parentData = \is_array($parentData) ? $parentData : unserialize($parentData);
```

Comment:
Quoting https://github.com/symfony/security-core/commit/f948fedde6c1b86d9a8b16d8016ce3376be20f74#diff-c5e43b78aa0cd524bf4965e55c5839b0e4eebc0afde1fd0ba45deb946d1734d9R187:

> There is no need to unserialize any entry in $data, they are already ready-to-use.

In that case, would you agree that this line is redundant and should be removed?